### PR TITLE
feat(destination): add PullRequestCommits data

### PR DESCRIPTION
**Pull request by saviogl**
## Description

Adding support for PullRequestCommits for Faros destination connector.

In conversation with @thomas-gerber, I've pointed out that the current way Faros CE calculates Lead Time for change is not applicable to all of our projects given that we are running an environment per branch git approach which includes multiple PRs in the development flow. 

With this data available we are able to craft queries to calculate Lead Time for changes based on the PR's commits **_created_at_** instead which gives us a more accurate and reliable metric.  

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

## Migration notes

> Describe migration notes if any

## Extra info

Airbyte Sync Example Log

```
2022-09-16 05:24:44 destination > Read 18429 messages
2022-09-16 05:24:44 destination > Read 17979 records
2022-09-16 05:24:44 destination > Processed 16449 records
2022-09-16 05:24:44 destination > Processed records by stream: {"github_source__github__assignees":171,"github_source__github__branches":5277,"github_source__github__collaborators":171,"github_source__github__comments":58,"github_source__github__commits":26,"github_source__github__events":248,"github_source__github__issue_comment_reactions":2,"github_source__github__issue_events":346,"github_source__github__issue_labels":37,"github_source__github__issues":114,"github_source__github__organizations":1,"github_source__github__pull_request_comment_reactions":7,"github_source__github__pull_request_commits":908,"github_source__github__pull_request_stats":114,"github_source__github__pull_requests":114,"github_source__github__releases":4,"github_source__github__repositories":2,"github_source__github__review_comments":35,"github_source__github__reviews":239,"github_source__github__tags":2856,"github_source__github__team_members":409,"github_source__github__team_memberships":409,"github_source__github__teams":51,"github_source__github__users":170,"github_source__github__workflow_runs":6204,"github_source__github__workflows":6}
2022-09-16 05:24:44 destination > Wrote 23100 records
2022-09-16 05:24:44 destination > Wrote records by model: {"cicd_Build":6204,"cicd_BuildCommitAssociation":6204,"cicd_Organization":1,"cicd_Pipeline":6,"cicd_Release":4,"cicd_ReleaseTagAssociation":4,"tms_Label":37,"tms_Project":2,"tms_TaskBoard":2,"tms_TaskBoardProjectRelationship":2,"tms_User":171,"vcs_Branch":5277,"vcs_BranchCommitAssociation":26,"vcs_Commit":26,"vcs_Membership":171,"vcs_Organization":1,"vcs_PullRequest":114,"vcs_PullRequestComment":35,"vcs_PullRequestCommit":908,"vcs_PullRequestReview":239,"vcs_PullRequest__Update":114,"vcs_Repository":2,"vcs_Tag":2856,"vcs_User":694}
2022-09-16 05:24:44 destination > Skipped 1530 records
2022-09-16 05:24:44 destination > Errored 0 records
```

### CODE REVIEW

File: destinations/airbyte-faros-destination/src/converters/github/pull_request_commits.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    const prCommit = record.record.data;
+    const res: DestinationRecord[] = [];
+
+    const repository: RepositoryKey = GitHubCommon.parseRepositoryKey(
+      prCommit.repository,
+      source
+    );
+
+    const commit = {
+      uid: prCommit.sha,
```
Comment (ypc-faros) [inline destinations/airbyte-faros-destination/src/converters/github/pull_request_commits.ts:26]: Please also add `sha: prCommit.sha` to the commit key.

The `sha` is part of the commit PK in Faros Saas (you had no way to know :) )

**Comment by ypc-faros** [comment]
Can you update the tests as well? They're not super involved. It'd be pretty much adding some records to the [test data](https://github.com/faros-ai/airbyte-connectors/blob/main/destinations/airbyte-faros-destination/test/resources/github/all-streams.log) and updating the write stats (number of records read/processed/written etc) in the github converter [test](https://github.com/faros-ai/airbyte-connectors/blob/main/destinations/airbyte-faros-destination/test/converters/github.test.ts) accordingly.

**Comment by saviogl** [comment]
@ypc-faros Let me know if that looks good 👍

**Review by ypc-faros — APPROVED** [review]
Thank you @saviogl for the contribution! 🥇

**Pull request merged**